### PR TITLE
brings in changes needed for FMS 2021.03

### LINKED ISSUE #761: updates needed to utilize FMS 2021.03
## Description
Updates needed to utilize FMS 2021.03

## Solution
PR #374 in fv3atm merged and then the PR to address this issue being merged

## Alternatives
N/A

## Related to
Directly reference any issues or PRs in this or other repositories that this is related to, and describe how they are related. Example:
- required to support ufs-community/ufs-weather-model/issue/760
- needed by PRs are yet to be crated

### PR DESCRIPTION
**Pull request by bensonr**
* new fv3atm (PR 374 3fef1582038a90eb117ada9b62cd546740b335a2)
  * updated stochastic_physics (PR 43 2dfaec8ca47cea9397c1c1e6e6836ce1acaa421a)

updates fv3_conf/compile_slurm.IN_gaea to use proper queue with ecflow

DOES NOT CHANGE ANSWERS

# PR Checklist

- [x] Ths PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [x] If new or updated input data is required by this PR, it is clearly stated in the text of the PR.

## Instructions: All subsequent sections of text should be filled in as appropriate.

The information provided below allows the code managers to understand the changes relevant to this PR, whether those changes are in the ufs-weather-model repository or in a subcomponent repository. Ufs-weather-model code managers will use the information provided to add any applicable labels, assign reviewers and place it in the Commit Queue. Once the PR is in the Commit Queue, it is the PR owner's responsiblity to keep the PR up-to-date with the develop branch of ufs-weather-model. 

## Description

Provide a detailed description of what this PR does. What bug does it fix, or what feature does it add? Is a change of answers expected from this PR? Are any library updates included in this PR (modulefiles etc.)?
This PR includes updates needed to utilize the latest FMS release 2021.03.  This is the initial step to bring in the latest FV3 "the dycore" updates required for advanced nesting capability that includes multiple and telescoping nests working within the regional system.

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues must always be created before starting work on a PR branch!) 
- fixes #761 
- fixes noaa-emc/fv3atm/issue/373

## Testing

How were these changes tested? What compilers / HPCs was it tested with? Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

- [x] hera.intel
- [x] hera.gnu
- [x] orion.intel
- [x] cheyenne.intel 
- [x] cheyenne.gnu
- [x] gaea.intel 
- [x] jet.intel
- [x] wcoss_cray
- [x] wcoss_dell_p3
- [ ] CI

## Dependencies

If testing this branch requires non-default branches in other repositories, list them. Those branches should have matching names (ideally).

This PR is using bensonr/fv3atm :: develop

Do PRs in upstream repositories need to be merged first?
If so add the "waiting for other repos" label and list the upstream PRs
- waiting on noaa-emc/fv3atm/pull/374

### CODE REVIEW

**Comment by MinsukJi-NOAA** [comment]
CI will not pass as fms library update is needed for the container.

File: tests/RegressionTests_gaea.intel.log [hunk 1]
```diff
@@ -1,3 +1,30 @@
+<<<<<<< HEAD
```
Comment (climbfuji) [inline tests/RegressionTests_gaea.intel.log]: This file has merge conflicts, but will be overwritten by auto-rt - therefore no problem.

File: tests/fv3_conf/compile_slurm.IN_gaea [hunk 2]
```diff
@@ -2,8 +2,9 @@
 #SBATCH -e err
 #SBATCH -o out
 #SBATCH --account=@[ACCNR]
-#SBATCH --qos=@[QUEUE]
-#SBATCH --clusters=@[PARTITION]
+##SBATCH --qos=@[QUEUE]
+#SBATCH --clusters=es
+#SBATCH --partition=eslogin
```
Comment (climbfuji) [inline tests/fv3_conf/compile_slurm.IN_gaea:7]: Thanks for making this change.

**Review by climbfuji — APPROVED** [review]
`.gitmodules` will need to be reverted and fv3atm submodule pointer updated prior to merging.

**Comment by BrianCurtis-NOAA** [comment]
Machine: cheyenne
Compiler: gnu
Job: RT
Repo location: /glade/scratch/dtcufsrt/autort/tests/auto/pr/716863137/20210823123009/ufs-weather-model
Please make changes and add the following label back:
cheyenne-gnu-RT

**Comment by BrianCurtis-NOAA** [comment]
Machine: hera
Compiler: gnu
Job: RT
Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/716863137/20210823183046/ufs-weather-model
Please make changes and add the following label back:
hera-gnu-RT

**Comment by BrianCurtis-NOAA** [comment]
Machine: cheyenne
Compiler: intel
Job: RT
Repo location: /glade/scratch/dtcufsrt/autort/tests/auto/pr/716863137/20210823121509/ufs-weather-model
Please make changes and add the following label back:
cheyenne-intel-RT

**Comment by MinsukJi-NOAA** [comment]
On Hera, a number of compile jobs failed. Some error messages are:
```
Error: Different CHARACTER lengths (7/4) in array constructor at (1)
/scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/716863137/20210823183046/ufs-weather-model/FV3/io/FV3GFS_io.F90:2406:130:

 2406 |        call register_restart_field(Phy_restart, trim(GFS_Restart%name3d(num)), var3_p, dimensions=(/'xaxis_1','yaxis_1','zaxis_1','Time'/), is_optional=.true.)
      |                                                                                                                                  1
Error: Different CHARACTER lengths (7/4) in array constructor at (1)
/scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/716863137/20210823183046/ufs-weather-model/FV3/io/FV3GFS_io.F90:2257:121:

 2257 |         call register_restart_field(Phy_restart, trim(GFS_Restart%name2d(num)), var2_p, dimensions=(/'xaxis_1','yaxis_1','Time'/), is_optional=.true.)
      |                                                                                                                         1
Error: Different CHARACTER lengths (7/4) in array constructor at (1)
/scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/716863137/20210823183046/ufs-weather-model/FV3/io/FV3GFS_io.F90:2261:131:

 2261 |         call register_restart_field(Phy_restart, trim(GFS_restart%name3d(num)), var3_p, dimensions=(/'xaxis_1','yaxis_1','zaxis_1','Time'/), is_optional=.true.)
      |                                                                                                                                   1
Error: Different CHARACTER lengths (7/4) in array constructor at (1)
/scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/716863137/20210823183046/ufs-weather-model/FV3/io/FV3GFS_io.F90:1915:107:

 1915 |          call register_restart_field(Sfc_restart, sfc_name2(num), var2_p, dimensions=(/'xaxis_1','yaxis_1','Time'/), is_optional=.true.)
      |                                                                                                           1
Error: Different CHARACTER lengths (7/4) in array constructor at (1)
/scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/716863137/20210823183046/ufs-weather-model/FV3/io/FV3GFS_io.F90:1917:108:

 1917 |          call register_restart_field(Sfc_restart, sfc_name2(num), var2_p, dimensions=(/'xaxis_1', 'yaxis_1', 'Time'/) )
      |                                                                                                            1
Error: Different CHARACTER lengths (7/4) in array constructor at (1)
/scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/716863137/20210823183046/ufs-weather-model/FV3/io/FV3GFS_io.F90:1925:108:

 1925 |          call register_restart_field(Sfc_restart, sfc_name2(num), var2_p, dimensions=(/'xaxis_1', 'yaxis_1', 'Time'/), is_optional=.not.mand)
      |                                                                                                            1
Error: Different CHARACTER lengths (7/4) in array constructor at (1)
/scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/716863137/20210823183046/ufs-weather-model/FV3/io/FV3GFS_io.F90:1932:108:

 1932 |          call register_restart_field(Sfc_restart, sfc_name2(num), var2_p, dimensions=(/'xaxis_1', 'yaxis_1', 'Time'/))
      |                                                                                                            1
Error: Different CHARACTER lengths (7/4) in array constructor at (1)
/scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/716863137/20210823183046/ufs-weather-model/FV3/io/FV3GFS_io.F90:1938:108:

 1938 |          call register_restart_field(Sfc_restart, sfc_name2(num), var2_p, dimensions=(/'xaxis_1', 'yaxis_1', 'Time'/), is_optional=.not.mand)

```

File: tests/RegressionTests_gaea.intel.log (hunk 1 above)
Comment (bensonr) [reply tests/RegressionTests_gaea.intel.log]: Yes, I did that on purpose as I knew it would be overwritten.

**Comment by bensonr** [comment]
@BrianCurtis-NOAA - I didn't change any labels and don't know what file or area these changes are needed.  Can you please enlighten me?

**Comment by bensonr** [comment]
@MinsukJi-NOAA - I will make the appropriate fixes for the gnu compiler.

**Comment by junwang-noaa** [comment]
@bensonr I think it means there are errors shown in the RT log. Thanks to @MinsukJi-NOAA for fixing the issue for gnu compiler.
@climbfuji Can you take a look at the RT log for Cheyenne intel?

**Comment by BrianCurtis-NOAA** [comment]
> @BrianCurtis-NOAA - I didn't change any labels and don't know what file or area these changes are needed. Can you please enlighten me?

I have the Automated testing scripts running through my github account. It's a good point to make that those unfamiliar with this fact will see those comments as you did. I'll look into this!

**Comment by BrianCurtis-NOAA** [comment]
Machine: hera
Compiler: intel
Job: RT
Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/716863137/20210823181508/ufs-weather-model
Please manually delete: /scratch1/NCEPDEV/stmp2/emc.nemspara/FV3_RT/rt_10129
Test cpld_debug 016 failed in check_result failed 
Test cpld_debug 016 failed in run_test failed 
Test fv3_rap 053 failed in check_result failed 
Test fv3_rap 053 failed in run_test failed 
Test fv3_hrrr 054 failed in check_result failed 
Test fv3_hrrr 054 failed in run_test failed 
Test fv3_rrfs_v1beta 055 failed in check_result failed 
Test fv3_rrfs_v1beta 055 failed in run_test failed 
Test fv3_rrfs_v1alpha 052 failed in check_result failed 
Test fv3_rrfs_v1alpha 052 failed in run_test failed 
Test cpld_ca 005 failed in check_result failed 
Test cpld_ca 005 failed in run_test failed 
Test cpld_decomp 004 failed in check_result failed 
Test cpld_decomp 004 failed in run_test failed 
Test cpld_2threads 003 failed in check_result failed 
Test cpld_2threads 003 failed in run_test failed 
Test cpld_control 001 failed in check_result failed 
Test cpld_control 001 failed in run_test failed 
Test fv3_gsd 051 failed in check_result failed 
Test fv3_gsd 051 failed in run_test failed 
Test cpld_control_wave 015 failed in check_result failed 
Test cpld_control_wave 015 failed in run_test failed 
Test cpld_control_c192 006 failed in check_result failed 
Test cpld_control_c192 006 failed in run_test failed 
Test regional_control 044 failed in check_result failed 
Test regional_control 044 failed in run_test failed 
Test regional_quilt_netcdf_parallel 049 failed in check_result failed 
Test regional_quilt_netcdf_parallel 049 failed in run_test failed 
Test datm_control_cfsr 089 failed in check_result failed 
Test datm_control_cfsr 089 failed in run_test failed 
Test datm_control_gefs 091 failed in check_result failed 
Test datm_control_gefs 091 failed in run_test failed 
Test datm_control_iau_gefs 092 failed in check_result failed 
Test datm_control_iau_gefs 092 failed in run_test failed 
Test datm_bulk_cfsr 093 failed in check_result failed 
Test datm_bulk_cfsr 093 failed in run_test failed 
Test control_c192 025 failed failed 
Test control_c192 025 failed in run_test failed 
Test datm_bulk_gefs 094 failed in check_result failed 
Test datm_bulk_gefs 094 failed in run_test failed 
Test datm_cdeps_multiple_files_cfsr 105 failed in check_result failed 
Test datm_cdeps_multiple_files_cfsr 105 failed in run_test failed 
Test cpld_control_c384 008 failed in check_result failed 
Test cpld_control_c384 008 failed in run_test failed 
Test datm_cdeps_control_gefs 100 failed in check_result failed 
Test datm_cdeps_control_gefs 100 failed in run_test failed 
Test datm_cdeps_control_cfsr 098 failed in check_result failed 
Test datm_cdeps_control_cfsr 098 failed in run_test failed 
Test datm_debug_cfsr 097 failed in check_result failed 
Test datm_debug_cfsr 097 failed in run_test failed 
Test datm_cdeps_bulk_cfsr 101 failed in check_result failed 
Test datm_cdeps_bulk_cfsr 101 failed in run_test failed 
Test datm_cdeps_bulk_gefs 102 failed in check_result failed 
Test datm_cdeps_bulk_gefs 102 failed in run_test failed 
Test cpld_bmark_v16_nsst 012 failed in check_result failed 
Test cpld_bmark_v16_nsst 012 failed in run_test failed 
Test cpld_bmark_v16 010 failed in check_result failed 
Test cpld_bmark_v16 010 failed in run_test failed 
Test cpld_bmark_wave_v16 013 failed in check_result failed 
Test cpld_bmark_wave_v16 013 failed in run_test failed 
Test datm_cdeps_debug_cfsr 106 failed in check_result failed 
Test datm_cdeps_debug_cfsr 106 failed in run_test failed 
Test datm_mx025_gefs 096 failed in check_result failed 
Test datm_mx025_gefs 096 failed in run_test failed 
Test datm_mx025_cfsr 095 failed in check_result failed 
Test datm_mx025_cfsr 095 failed in run_test failed 
Test datm_cdeps_mx025_cfsr 103 failed in check_result failed 
Test datm_cdeps_mx025_cfsr 103 failed in run_test failed 
Test datm_cdeps_mx025_gefs 104 failed in check_result failed 
Test datm_cdeps_mx025_gefs 104 failed in run_test failed 
Test cpld_bmark_wave_v16_p7b 014 failed in check_result failed 
Test cpld_bmark_wave_v16_p7b 014 failed in run_test failed 
Please make changes and add the following label back:
hera-intel-RT

**Comment by DusanJovic-NOAA** [comment]
> Machine: hera
> Compiler: intel
> Job: RT
> Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/716863137/20210823181508/ufs-weather-model
> Please manually delete: /scratch1/NCEPDEV/stmp2/emc.nemspara/FV3_RT/rt_10129
> Test cpld_debug 016 failed in check_result failed
> Test cpld_debug 016 failed in run_test failed
> Test fv3_rap 053 failed in check_result failed
> Test fv3_rap 053 failed in run_test failed
> Test fv3_hrrr 054 failed in check_result failed
> Test fv3_hrrr 054 failed in run_test failed
> Test fv3_rrfs_v1beta 055 failed in check_result failed
> Test fv3_rrfs_v1beta 055 failed in run_test failed
> Test fv3_rrfs_v1alpha 052 failed in check_result failed
> Test fv3_rrfs_v1alpha 052 failed in run_test failed
> Test cpld_ca 005 failed in check_result failed
> Test cpld_ca 005 failed in run_test failed
> Test cpld_decomp 004 failed in check_result failed
> Test cpld_decomp 004 failed in run_test failed
> Test cpld_2threads 003 failed in check_result failed
> Test cpld_2threads 003 failed in run_test failed
> Test cpld_control 001 failed in check_result failed
> Test cpld_control 001 failed in run_test failed
> Test fv3_gsd 051 failed in check_result failed
> Test fv3_gsd 051 failed in run_test failed
> Test cpld_control_wave 015 failed in check_result failed
> Test cpld_control_wave 015 failed in run_test failed
> Test cpld_control_c192 006 failed in check_result failed
> Test cpld_control_c192 006 failed in run_test failed
> Test regional_control 044 failed in check_result failed
> Test regional_control 044 failed in run_test failed
> Test regional_quilt_netcdf_parallel 049 failed in check_result failed
> Test regional_quilt_netcdf_parallel 049 failed in run_test failed
> Test datm_control_cfsr 089 failed in check_result failed
> Test datm_control_cfsr 089 failed in run_test failed
> Test datm_control_gefs 091 failed in check_result failed
> Test datm_control_gefs 091 failed in run_test failed
> Test datm_control_iau_gefs 092 failed in check_result failed
> Test datm_control_iau_gefs 092 failed in run_test failed
> Test datm_bulk_cfsr 093 failed in check_result failed
> Test datm_bulk_cfsr 093 failed in run_test failed
> Test control_c192 025 failed failed
> Test control_c192 025 failed in run_test failed
> Test datm_bulk_gefs 094 failed in check_result failed
> Test datm_bulk_gefs 094 failed in run_test failed
> Test datm_cdeps_multiple_files_cfsr 105 failed in check_result failed
> Test datm_cdeps_multiple_files_cfsr 105 failed in run_test failed
> Test cpld_control_c384 008 failed in check_result failed
> Test cpld_control_c384 008 failed in run_test failed
> Test datm_cdeps_control_gefs 100 failed in check_result failed
> Test datm_cdeps_control_gefs 100 failed in run_test failed
> Test datm_cdeps_control_cfsr 098 failed in check_result failed
> Test datm_cdeps_control_cfsr 098 failed in run_test failed
> Test datm_debug_cfsr 097 failed in check_result failed
> Test datm_debug_cfsr 097 failed in run_test failed
> Test datm_cdeps_bulk_cfsr 101 failed in check_result failed
> Test datm_cdeps_bulk_cfsr 101 failed in run_test failed
> Test datm_cdeps_bulk_gefs 102 failed in check_result failed
> Test datm_cdeps_bulk_gefs 102 failed in run_test failed
> Test cpld_bmark_v16_nsst 012 failed in check_result failed
> Test cpld_bmark_v16_nsst 012 failed in run_test failed
> Test cpld_bmark_v16 010 failed in check_result failed
> Test cpld_bmark_v16 010 failed in run_test failed
> Test cpld_bmark_wave_v16 013 failed in check_result failed
> Test cpld_bmark_wave_v16 013 failed in run_test failed
> Test datm_cdeps_debug_cfsr 106 failed in check_result failed
> Test datm_cdeps_debug_cfsr 106 failed in run_test failed
> Test datm_mx025_gefs 096 failed in check_result failed
> Test datm_mx025_gefs 096 failed in run_test failed
> Test datm_mx025_cfsr 095 failed in check_result failed
> Test datm_mx025_cfsr 095 failed in run_test failed
> Test datm_cdeps_mx025_cfsr 103 failed in check_result failed
> Test datm_cdeps_mx025_cfsr 103 failed in run_test failed
> Test datm_cdeps_mx025_gefs 104 failed in check_result failed
> Test datm_cdeps_mx025_gefs 104 failed in run_test failed
> Test cpld_bmark_wave_v16_p7b 014 failed in check_result failed
> Test cpld_bmark_wave_v16_p7b 014 failed in run_test failed
> Please make changes and add the following label back:
> hera-intel-RT

For example, regional_control test failed because atmos_4xdaily.nc file is different compared to the baseline:

```
-rw-r--r-- 1 emc.nemspara stmp 40676312 Aug 20 18:52 /scratch1/NCEPDEV/nems/emc.nemspara/RT/NEMSfv3gfs/develop-20210820/INTEL/fv3_regional_control/atmos_4xdaily.nc
-rw-r--r-- 1 emc.nemspara nems 40674592 Aug 23 18:47 /scratch1/NCEPDEV/stmp2/emc.nemspara/FV3_RT/rt_10129/regional_control/atmos_4xdaily.nc

```

**Comment by DusanJovic-NOAA** [comment]
> > Machine: hera
> > Compiler: intel
> > Job: RT
> > Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/716863137/20210823181508/ufs-weather-model
> > Please manually delete: /scratch1/NCEPDEV/stmp2/emc.nemspara/FV3_RT/rt_10129
> > Test cpld_debug 016 failed in check_result failed
> > Test cpld_debug 016 failed in run_test failed
> > Test fv3_rap 053 failed in check_result failed
> > Test fv3_rap 053 failed in run_test failed
> > Test fv3_hrrr 054 failed in check_result failed
> > Test fv3_hrrr 054 failed in run_test failed
> > Test fv3_rrfs_v1beta 055 failed in check_result failed
> > Test fv3_rrfs_v1beta 055 failed in run_test failed
> > Test fv3_rrfs_v1alpha 052 failed in check_result failed
> > Test fv3_rrfs_v1alpha 052 failed in run_test failed
> > Test cpld_ca 005 failed in check_result failed
> > Test cpld_ca 005 failed in run_test failed
> > Test cpld_decomp 004 failed in check_result failed
> > Test cpld_decomp 004 failed in run_test failed
> > Test cpld_2threads 003 failed in check_result failed
> > Test cpld_2threads 003 failed in run_test failed
> > Test cpld_control 001 failed in check_result failed
> > Test cpld_control 001 failed in run_test failed
> > Test fv3_gsd 051 failed in check_result failed
> > Test fv3_gsd 051 failed in run_test failed
> > Test cpld_control_wave 015 failed in check_result failed
> > Test cpld_control_wave 015 failed in run_test failed
> > Test cpld_control_c192 006 failed in check_result failed
> > Test cpld_control_c192 006 failed in run_test failed
> > Test regional_control 044 failed in check_result failed
> > Test regional_control 044 failed in run_test failed
> > Test regional_quilt_netcdf_parallel 049 failed in check_result failed
> > Test regional_quilt_netcdf_parallel 049 failed in run_test failed
> > Test datm_control_cfsr 089 failed in check_result failed
> > Test datm_control_cfsr 089 failed in run_test failed
> > Test datm_control_gefs 091 failed in check_result failed
> > Test datm_control_gefs 091 failed in run_test failed
> > Test datm_control_iau_gefs 092 failed in check_result failed
> > Test datm_control_iau_gefs 092 failed in run_test failed
> > Test datm_bulk_cfsr 093 failed in check_result failed
> > Test datm_bulk_cfsr 093 failed in run_test failed
> > Test control_c192 025 failed failed
> > Test control_c192 025 failed in run_test failed
> > Test datm_bulk_gefs 094 failed in check_result failed
> > Test datm_bulk_gefs 094 failed in run_test failed
> > Test datm_cdeps_multiple_files_cfsr 105 failed in check_result failed
> > Test datm_cdeps_multiple_files_cfsr 105 failed in run_test failed
> > Test cpld_control_c384 008 failed in check_result failed
> > Test cpld_control_c384 008 failed in run_test failed
> > Test datm_cdeps_control_gefs 100 failed in check_result failed
> > Test datm_cdeps_control_gefs 100 failed in run_test failed
> > Test datm_cdeps_control_cfsr 098 failed in check_result failed
> > Test datm_cdeps_control_cfsr 098 failed in run_test failed
> > Test datm_debug_cfsr 097 failed in check_result failed
> > Test datm_debug_cfsr 097 failed in run_test failed
> > Test datm_cdeps_bulk_cfsr 101 failed in check_result failed
> > Test datm_cdeps_bulk_cfsr 101 failed in run_test failed
> > Test datm_cdeps_bulk_gefs 102 failed in check_result failed
> > Test datm_cdeps_bulk_gefs 102 failed in run_test failed
> > Test cpld_bmark_v16_nsst 012 failed in check_result failed
> > Test cpld_bmark_v16_nsst 012 failed in run_test failed
> > Test cpld_bmark_v16 010 failed in check_result failed
> > Test cpld_bmark_v16 010 failed in run_test failed
> > Test cpld_bmark_wave_v16 013 failed in check_result failed
> > Test cpld_bmark_wave_v16 013 failed in run_test failed
> > Test datm_cdeps_debug_cfsr 106 failed in check_result failed
> > Test datm_cdeps_debug_cfsr 106 failed in run_test failed
> > Test datm_mx025_gefs 096 failed in check_result failed
> > Test datm_mx025_gefs 096 failed in run_test failed
> > Test datm_mx025_cfsr 095 failed in check_result failed
> > Test datm_mx025_cfsr 095 failed in run_test failed
> > Test datm_cdeps_mx025_cfsr 103 failed in check_result failed
> > Test datm_cdeps_mx025_cfsr 103 failed in run_test failed
> > Test datm_cdeps_mx025_gefs 104 failed in check_result failed
> > Test datm_cdeps_mx025_gefs 104 failed in run_test failed
> > Test cpld_bmark_wave_v16_p7b 014 failed in check_result failed
> > Test cpld_bmark_wave_v16_p7b 014 failed in run_test failed
> > Please make changes and add the following label back:
> > hera-intel-RT
> 
> For example, regional_control test failed because atmos_4xdaily.nc file is different compared to the baseline:
> 
> ```
> -rw-r--r-- 1 emc.nemspara stmp 40676312 Aug 20 18:52 /scratch1/NCEPDEV/nems/emc.nemspara/RT/NEMSfv3gfs/develop-20210820/INTEL/fv3_regional_control/atmos_4xdaily.nc
> -rw-r--r-- 1 emc.nemspara nems 40674592 Aug 23 18:47 /scratch1/NCEPDEV/stmp2/emc.nemspara/FV3_RT/rt_10129/regional_control/atmos_4xdaily.nc
> ```


I see differences in type and some attributes of some variables between these two files:

```
$ diff <(ncdump -h /scratch1/NCEPDEV/stmp2/emc.nemspara/FV3_RT/rt_10129/regional_control/atmos_4xdaily.nc) <(ncdump -h /scratch1/NCEPDEV/nems/emc.nemspara/RT/NEMSfv3gfs/develop-20210820/INTEL/fv3_regional_control/atmos_4xdaily.nc)
7c7,8
<       float grid_xt(grid_xt) ;
---
>       double grid_xt(grid_xt) ;
>               grid_xt:domain_decomposition = 1, 210, 1, 210 ;
11c12,13
<       float grid_yt(grid_yt) ;
---
>       double grid_yt(grid_yt) ;
>               grid_yt:domain_decomposition = 1, 192, 1, 192 ;
15c17
<       float time(time) ;
---
>       double time(time) ;

```

**Comment by bensonr** [comment]
Updates for the gnu compiler (FV3GFS_io.F90) have been pushed to bensonr/fv3atm and the hash for FV3 (fv3atm) updated.  Tests are running on hera for the gnu compiler (RT_COMPILER=gnu and 'rt.sh -l rt_gnu.conf')

**Comment by climbfuji** [comment]
> @bensonr I think it means there are errors shown in the RT log. Thanks to @MinsukJi-NOAA for fixing the issue for gnu compiler.
> @climbfuji Can you take a look at the RT log for Cheyenne intel?

Not sure what was going on there, maybe disk space issues. Will run the tests manually for cheyenne.intel now that the gnu bugfixes for the I/O layer have been pushed.

**Comment by climbfuji** [comment]
> > @bensonr I think it means there are errors shown in the RT log. Thanks to @MinsukJi-NOAA for fixing the issue for gnu compiler.
> > @climbfuji Can you take a look at the RT log for Cheyenne intel?
> 
> Not sure what was going on there, maybe disk space issues. Will run the tests manually for cheyenne.intel now that the gnu bugfixes for the I/O layer have been pushed.

I just kicked off `rt.sh` on Cheyenne/Intel - will take a while, Cheyenne is busy today.

**Comment by bensonr** [comment]
I've run hera.gnu and hera.intel by hand.  

For hera.gnu, 28 of 33 tests pass.  Of the five that fail, three only have atmos_4xdaily.nc differences.  The other two are the cpld_* cases and both are completely different.

For hera.intel,  67 of 102 tests pass.  Of the 35 that fail, seven only have atmos_4xdaily.nc differences.  The other 28 are the full set of cpld and datm tests with different answers.  What's strange is that all of these tests pass with Intel on Orion, Jet, and Gaea with Intel.

**Comment by climbfuji** [comment]
> I've run hera.gnu and hera.intel by hand.
> 
> For hera.gnu, 28 of 33 tests pass. Of the five that fail, three only have atmos_4xdaily.nc differences. The other two are the cpld_* cases and both are completely different.
> 
> For hera.intel, 67 of 102 tests pass. Of the 35 that fail, seven only have atmos_4xdaily.nc differences. The other 28 are the full set of cpld and datm tests with different answers. What's strange is that all of these tests pass with Intel on Orion, Jet, and Gaea with Intel.

@DusanJovic-NOAA and I suspect that this has to do with how FMS was compiled. On Gaea, for instance, I compiled FMS with exactly the same compiler flags (in particular no AVX2, as for the previous version) and CPP flags - there, all tests passed.

**Comment by DusanJovic-NOAA** [comment]
The following tests failed on wcoss_dell_p3:
```
$ cat fail_test 
cpld_control 001 failed in check_result
cpld_restart 002 failed in check_result
cpld_2threads 003 failed in check_result
cpld_decomp 004 failed in check_result
cpld_ca 005 failed in check_result
cpld_control_c192 006 failed in check_result
cpld_restart_c192 007 failed in check_result
cpld_control_c384 008 failed in check_result
cpld_restart_c384 009 failed in check_result
cpld_bmark_v16 010 failed in check_result
cpld_restart_bmark_v16 011 failed in check_result
cpld_bmark_v16_nsst 012 failed in check_result
cpld_bmark_wave_v16 013 failed in check_result
cpld_bmark_wave_v16_p7b 014 failed in check_result
cpld_control_wave 015 failed in check_result
cpld_debug 016 failed in check_result
regional_control 044 failed in check_result
regional_restart 045 failed in check_result
regional_quilt_netcdf_parallel 049 failed in check_result
fv3_gsd 051 failed in check_result
fv3_rrfs_v1alpha 052 failed in check_result
fv3_rap 053 failed in check_result
fv3_hrrr 054 failed in check_result
fv3_rrfs_v1beta 055 failed in check_result
datm_control_cfsr 089 failed in check_result
datm_restart_cfsr 090 failed in check_result
datm_control_gefs 091 failed in check_result
datm_control_iau_gefs 092 failed in check_result
datm_bulk_cfsr 093 failed in check_result
datm_bulk_gefs 094 failed in check_result
datm_mx025_cfsr 095 failed in check_result
datm_mx025_gefs 096 failed in check_result
datm_debug_cfsr 097 failed in check_result
datm_cdeps_control_cfsr 098 failed in check_result
datm_cdeps_restart_cfsr 099 failed in check_result
datm_cdeps_control_gefs 100 failed in check_result
datm_cdeps_bulk_cfsr 101 failed in check_result
datm_cdeps_bulk_gefs 102 failed in check_result
datm_cdeps_mx025_cfsr 103 failed in check_result
datm_cdeps_mx025_gefs 104 failed in check_result
datm_cdeps_multiple_files_cfsr 105 failed in check_result
datm_cdeps_debug_cfsr 106 failed in check_result

```

**Review by junwang-noaa — APPROVED** [review]

**Review by DusanJovic-NOAA — APPROVED** [review]

**Pull request merged**